### PR TITLE
Task-41419_42410: Implement trendMicro malware detection connector and implement malwareDetection files clean

### PR DESCRIPTION
**Pull request by azayati**
* This PR implements trendMicro malware detection connector, this connector is defined via the configuration as default MalwareDetectionConnector with the property exo.malwareDetection.connector.trendMicro.isDefault=true 
* In order to call trendMicro API we need these properties:
> exo.malwareDetection.connector.trendMicro.api.host
> exo.malwareDetection.connector.trendMicro.api.userName
> exo.malwareDetection.connector.trendMicro.api.password
> exo.malwareDetection.connector.trendMicro.api.listEvents.delta (Optional, 15 min by default, this represents the delta between the date of malware upload and the date of the corresponding event detection, expressed in ms)

### CODE REVIEW

File: component/service/src/main/java/org/exoplatform/social/service/malwareDetection/connector/TrendMicroMalwareDetectionConnector.java [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+                        MALWARE_DETECTION_FEATURE,
+                        TRENDMICRO_LOGIN_OPERATION,
+                        trendmicroHost,
+                        trendmicroUserName,
+                        endTime - startTime);
+            return response.toString();
+          }
+        } 
+      } catch (Exception e) {
+        LOGGER.error("Error when trying to login trendMicro with host={} and userName={}", trendmicroHost, trendmicroUserName, e);
```
Comment (rdenarie) [inline component/service/src/main/java/org/exoplatform/social/service/malwareDetection/connector/TrendMicroMalwareDetectionConnector.java]: Should we use log format with service={} operation={} and status=ko ?

File: component/service/src/main/java/org/exoplatform/social/service/malwareDetection/connector/TrendMicroMalwareDetectionConnector.java [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+                         MALWARE_DETECTION_FEATURE,
+                         TRENDMICRO_LIST_EVENTS_OPERATION,
+                         trendmicroHost,
+                         sid,
+                         endTime - startTime);
+             return response.toString();
+           }
+        }
+      } catch (Exception e) {
+        LOGGER.error("Error when trying to get trendmicro events list with host={} and sID={}", trendmicroHost, sid, e);
```
Comment (rdenarie) [inline component/service/src/main/java/org/exoplatform/social/service/malwareDetection/connector/TrendMicroMalwareDetectionConnector.java]: Should we use log format with service={} operation={} and status=ko

File: component/service/src/main/java/org/exoplatform/social/service/malwareDetection/connector/TrendMicroMalwareDetectionConnector.java [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+          long endTime = System.currentTimeMillis();
+          LOGGER.info("service={} operation={} parameters=\"host:{}\" \"sID:{}\" status=ok " + "duration_ms={}",
+                      MALWARE_DETECTION_FEATURE,
+                      TRENDMICRO_LOGOUT_OPERATION,
+                      trendmicroHost,
+                      sid,
+                      endTime - startTime);
+        }
+      } catch (Exception e) {
+        LOGGER.error("Error when trying to logout trendMicro with host={} and sID={}", trendmicroHost, sid, e);
```
Comment (rdenarie) [inline component/service/src/main/java/org/exoplatform/social/service/malwareDetection/connector/TrendMicroMalwareDetectionConnector.java]: Should we use log format with service={} operation={} and status=ko

**Review by rdenarie — CHANGES_REQUESTED** [review]

File: component/service/src/main/java/org/exoplatform/social/service/malwareDetection/connector/MalwareDetectionFilesConnector.java [hunk 4]
```diff
@@ -50,24 +52,38 @@ public MalwareDetectionFilesConnector(InitParams initParams, FileService fileSer
         }
       }
     } catch (Exception e) {
-      LOGGER.error("Error when trying to get the infected item informations from fileChecksum={}", fileChecksum, e);
+      long endTime = System.currentTimeMillis();
+      LOGGER.error("service={} operation={} parameters=\"fileChecksum:{}\" status=ko duration_ms={} error_msg=Error when trying to get the infected items:{}",
```
Comment (rdenarie) [inline component/service/src/main/java/org/exoplatform/social/service/malwareDetection/connector/MalwareDetectionFilesConnector.java]:
```suggestion
      LOGGER.error("service={} operation={} parameters=\"fileChecksum:{}\" status=ko duration_ms={} error_msg=\"Error when trying to get the infected items:{}\"",
```
https://community.exoplatform.com/portal/g/:spaces:exo_itop/exo_itop/wiki/SPEC_-_logging_for_monitoring

**Review by rdenarie — CHANGES_REQUESTED** [review]

File: component/service/src/main/java/org/exoplatform/social/service/malwareDetection/connector/MalwareDetectionFilesConnector.java (hunk 4 above)
Comment (azayati) [reply component/service/src/main/java/org/exoplatform/social/service/malwareDetection/connector/MalwareDetectionFilesConnector.java]: updated

File: component/service/src/main/java/org/exoplatform/social/service/malwareDetection/connector/TrendMicroMalwareDetectionConnector.java [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+                        MALWARE_DETECTION_FEATURE,
+                        TRENDMICRO_LOGIN_OPERATION,
+                        trendmicroHost,
+                        trendmicroUserName,
+                        endTime - startTime);
+            return response.toString();
+          }
+        }
+        long endTime = System.currentTimeMillis();
+        LOGGER.error("remote_service={} operation={} parameters=\"host:{}\" \"userName:{}\" status=ko duration_ms={} error_msg=\"Error when trying to login trendMicro\"",
```
Comment (rdenarie) [inline component/service/src/main/java/org/exoplatform/social/service/malwareDetection/connector/TrendMicroMalwareDetectionConnector.java]: should we put response code in error message ?

File: component/service/src/main/java/org/exoplatform/social/service/malwareDetection/connector/TrendMicroMalwareDetectionConnector.java [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+                         MALWARE_DETECTION_FEATURE,
+                         TRENDMICRO_LIST_EVENTS_OPERATION,
+                         trendmicroHost,
+                         sid,
+                         endTime - startTime);
+             return response.toString();
+           }
+        }
+        long endTime = System.currentTimeMillis();
+        LOGGER.error("remote_service={} operation={} parameters=\"host:{}\" \"sID:{}\" status=ko duration_ms={} error_msg=\"Error when trying to get trendmicro events list\"",
```
Comment (rdenarie) [inline component/service/src/main/java/org/exoplatform/social/service/malwareDetection/connector/TrendMicroMalwareDetectionConnector.java]: should we put reponse code in error message

File: component/service/src/main/java/org/exoplatform/social/service/malwareDetection/connector/TrendMicroMalwareDetectionConnector.java [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+        if (trendmicroLogoutUrlConnection.getResponseCode() == HttpURLConnection.HTTP_OK) { //success
+          LOGGER.info("remote_service={} operation={} parameters=\"host:{}\" \"sID:{}\" status=ok duration_ms={}",
+                      MALWARE_DETECTION_FEATURE,
+                      TRENDMICRO_LOGOUT_OPERATION,
+                      trendmicroHost,
+                      sid,
+                      endTime - startTime);
+        }
+        else {
+          LOGGER.error("remote_service={} operation={} parameters=\"host:{}\" \"sID:{}\" status=ko duration_ms={} error_msg=\"Error when trying to logout trendMicro\"",
```
Comment (rdenarie) [inline component/service/src/main/java/org/exoplatform/social/service/malwareDetection/connector/TrendMicroMalwareDetectionConnector.java]: should we put response code in error message

**Review by rdenarie — CHANGES_REQUESTED** [review]

**Review by rdenarie — APPROVED** [review]

**Pull request merged**